### PR TITLE
operators: stake management chapter

### PR DESCRIPTION
**Pull request by matevz**
Docs part of PR for https://github.com/oasislabs/oasis-core/issues/2275

This PR adds
* *Stake management in details using CLI* chapter for the node operators.

Merge after https://github.com/oasislabs/oasis-core/pull/2293 is merged.

### CODE REVIEW

File: src/operators/stake-management.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+```
+
+In our case `/tmp/oasis-net-runner236357163/net-runner/network/client-0/internal.sock`
+is the path to the internal unix socket we will use to perform staking
+operations.
+
+First, let's see what is the native token of our platform:
+
+```bash
+$ ./go/oasis-node/oasis-node stake info \
```
Comment (david-yan) [inline src/operators/stake-management.md]: When I ran this locally, I got the following error at the end of the output:
```
ts=2019-11-05T21:17:24.8275434Z level=warn module=cmd/stake caller=stake.go:68 msg="failed to query staking threshold(s)" err="rpc error: code = Unknown desc = staking: invalid threshold" attempt=1
```
I'm not sure if this is a bug, if I'm doing something wrong, of if it's working as intended, but we should either fix the error or provide a fix in the docs.

File: src/operators/stake-management.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+TODO: Instructions for signing transactions using HSM once it is implemented.
+
+The generation and signing of the transaction can be done offline. To submit our
+transaction however, we need to connect to our local Oasis node as follows:
+```bash
+./go/oasis-node/oasis-node stake account submit \
+ -a unix:/tmp/oasis-net-runner236357163/net-runner/network/client-0/internal.sock \
+ --stake.account.id 4ea5328f943ef6f66daaed74cb0e99c3b1c45f76307b425003dbc7cb3638ed35\
+ --stake.transaction.file b.json
```
Comment (david-yan) [inline src/operators/stake-management.md]: I'm getting the following when running this:
```
Error: unknown flag: --stake.transaction.file
```

File: src/operators/stake-management.md (hunk 1 above)
Comment (matevz) [reply src/operators/stake-management.md]: Correct, this is fixed by https://github.com/oasislabs/oasis-core/pull/2293 The problem is that thresholds must be at least set to 0. In current master, they aren't there at all.

File: src/operators/stake-management.md (hunk 2 above)
Comment (matevz) [reply src/operators/stake-management.md]: Yeah, that was a bug in master. Flag was added in https://github.com/oasislabs/oasis-core/pull/2293. Before that you couldn't submit anything.

File: src/operators/stake-management.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+```
+
+Shortly, three compute nodes, storage nodes and a key manager will start. If you
+track the output on the terminal, it should contain a similar line to this:
+
+```bash
+level=info module=net-runner caller=root.go:145 ts=2019-10-25T10:54:09.198642162Z msg="client node socket available" path=/tmp/oasis-net-runner236357163/net-runner/network/client-0/internal.sock
+```
+
+In our case `/tmp/oasis-net-runner236357163/net-runner/network/client-0/internal.sock`
```
Comment (david-yan) [inline src/operators/stake-management.md]: I'd suggest adding something here like
```
export SOCKET=/tmp/oasis-net-runner236357163/net-runner/network/client-0/internal.sock
```
and using the environment variable in the future calls. This makes it a lot easier for the user to copy and paste commands, and is what I ended up doing when running everything.

File: src/operators/stake-management.md [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+is the path to the internal unix socket we will use to perform staking
+operations.
+
+First, let's see what is the native token of our platform:
+
+```bash
+$ ./go/oasis-node/oasis-node stake info \
+ -a unix:/tmp/oasis-net-runner236357163/net-runner/network/client-0/internal.sock
```
Comment (david-yan) [inline src/operators/stake-management.md]:
```suggestion
 -a unix:$SOCKET
```

File: src/operators/stake-management.md [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+There is a native token named `Buffycoin` with symbol `BUF` in our test environment
+with total supply of 200 billion. All tokens are allocated in their respective
+accounts and no tokens are in the *common pool*. Finally we see no staking
+thresholds for any node kind (entity, validator, compute, storage).
+
+Let's list all accounts with positive balance:
+
+```bash
+$ ./go/oasis-node/oasis-node stake list \
+ -a unix:/tmp/oasis-net-runner236357163/net-runner/network/client-0/internal.sock
```
Comment (david-yan) [inline src/operators/stake-management.md]:
```suggestion
 -a unix:$SOCKET
```

File: src/operators/stake-management.md (hunk 3 above)
Comment (matevz) [reply src/operators/stake-management.md]: Good idea! I used `export ADDR=unix:...` and then `-a $ADDR` flag.

**Comment by ravenac95** [comment]
I actually think it may be best to merge this with the operator docs we are preparing to merge into the public docs here. The content here is great! I just want to make sure that we align some of the details with the operator docs. I will take a look at this tomorrow

File: src/operators/stake-management.md [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+
+```bash
+$ ./go/oasis-net-runner/oasis-net-runner \
+ --net.node.binary go/oasis-node/oasis-node \
+ --net.runtime.binary target/debug/simple-keyvalue \
+ --net.runtime.loader target/debug/oasis-core-runtime-loader \
+ --net.keymanager.binary target/debug/oasis-core-keymanager-runtime
+```
+
+Shortly, three compute nodes, storage nodes and a key manager will start. If you
```
Comment (ravenac95) [inline src/operators/stake-management.md]: For what we're releasing in the public testnet, we won't actually be running anything other than Validators. I wonder if we should just have people mess with staking on the public testnet.

**Comment by matevz** [comment]
We have new two fees parameters which need to be added.

File: src/operators/stake-management.md (hunk 6 above)
Comment (matevz) [reply src/operators/stake-management.md]: Yeah, at the time of writing the public testnet wasn't there yet. Do we have instructions how to connect to it and I will just point the reader from here?

Comment (matevz) [reply src/operators/stake-management.md]: I added testnet notes on various locations. Can I go ahead and merge it?

**Comment by matevz** [comment]
@ravenac95 I think the docs are now on-par with the rest of the docs. Can I go ahead and merge it?

File: src/operators/stake-management.md (hunk 6 above)
Comment (matevz) [reply src/operators/stake-management.md]: I added notes for testnet on number of locations in the new chapter. Should look good now.

**Pull request merged**